app/main.py

```diff
@@ -2,5 +1,21 @@
+from typing import Callable, Any
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    calculated_data = []
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        for value in args:
+            if isinstance(value, (list, set, dict)):
+                return None
+        for args_save, kwargs_save, result_save in calculated_data:
+            if args_save == args and kwargs_save == kwargs:
+                print("Getting from cache")
+                return result_save
+        result = func(*args, **kwargs)
+        calculated_data.append((args, kwargs, result))
```

Comment:
Storing `args` and `kwargs` directly in `calculated_data` can lead to issues if they contain mutable objects. Consider converting them to immutable types (e.g., using `tuple` for lists) before storing.